runbotics/runbotics-desktop/src/core/bpm/runtime/runtime.service.ts

```diff
@@ -588,6 +588,17 @@ export class RuntimeService implements OnApplicationBootstrap, OnModuleDestroy {
                             `[${processInstanceId}] [${executionId}] [${script}] Error running desktop action`,
                             (e as Error)?.message
                         );
+                        this.purgeEngine(processInstanceId);
+                        const processInstance = this.processInstances[processInstanceId];
+                        this.processEventBus.publish({
+                            processInstanceId: processInstanceId,
+                            eventType: ProcessInstanceStatus.ERRORED,
+                            processInstance: {
+                                ...processInstance,
+                                status: ProcessInstanceStatus.ERRORED,
+                                error: `Inputs in next step: ${(e as Error)?.message}`,
+                            },
+                        });
```

Comment:
we also publish error in the listener, so it will be done twice. If this code is mandatory for catching variables/services resolve issues then you should create some kind of if guard to send only one message